packages/data-stores/src/reader/queries/use-subscription-manager-site-subscriptions-query.ts

```diff
@@ -0,0 +1,71 @@
+import { useQuery } from 'react-query';
+import { callApi } from '../helpers';
+import { useIsLoggedIn, useIsQueryEnabled } from '../hooks';
+import type { SiteSubscription } from '../types';
+
+type SubscriptionManagerSiteSubscriptions = {
+	subscriptions: SiteSubscription[];
+	page: number;
+	total_subscriptions: number;
+};
+
+type SubscriptionManagerSiteSubscriptionsQueryProps = {
+	filter?: ( item?: SiteSubscription ) => boolean;
+	sort?: ( a?: SiteSubscription, b?: SiteSubscription ) => number;
+	limit?: number;
+};
+
+const callFollowingEndPoint = async (
+	page: number,
+	limit: number,
+	isLoggedIn: boolean
+): Promise< SiteSubscription[] > => {
+	const data = [];
+
+	const incoming = await callApi< SubscriptionManagerSiteSubscriptions >( {
+		path: `/read/following/mine?limit=${ limit }&page=${ page }`,
+		isLoggedIn,
+		apiVersion: '1.2',
+	} );
+
+	if ( incoming && incoming.subscriptions ) {
+		data.push( ...incoming.subscriptions );
+	}
+
+	if ( incoming.page * limit < incoming.total_subscriptions ) {
+		const next = await callFollowingEndPoint( page + 1, limit, isLoggedIn );
+		data.push( ...next );
+	}
+
+	return data;
+};
+
+const defaultFilter = () => true;
+const defaultSort = () => 0;
+
+const useSubscriptionManagerSiteSubscriptionsQuery = ( {
+	filter,
+	sort,
+	limit = 100,
+}: SubscriptionManagerSiteSubscriptionsQueryProps ) => {
```

Comment:
I would say that if we want to let the callback be used without any parameters as well, I think we will need to set the default values here:
```suggestion
const useSubscriptionManagerSiteSubscriptionsQuery = ( {
	filter = defaultFilter,
	sort = defaultSort,
	limit = 100,
}: SubscriptionManagerSiteSubscriptionsQueryProps = {} ) => {
```